cmd/sourced/compose/workdir/factory.go

```diff
@@ -143,78 +136,137 @@ func initialize(dirName string, subPath string, envf envFile) (*Workdir, error)
+	f := envFile{
+		ComposeProjectName: fmt.Sprintf("srcd-%s", dirName),
+
+		GitbaseVolumeType:   "bind",
+		GitbaseVolumeSource: repoDir,
 	}
+	f.addResourceLimits()
+
+	return f
+}
+
+func newOrgEnvFile(dirName string, orgs []string, token string, withForks bool) envFile {
+	f := envFile{
+		ComposeProjectName: fmt.Sprintf("srcd-%s", dirName),
+
+		GitbaseVolumeType:   "volume",
+		GitbaseVolumeSource: "gitbase_repositories",
+		GitbaseSiva:         true,
 
-	noForks := "true"
-	if f.WithForks {
-		noForks = "false"
+		GithubOrganizations: orgs,
+		GithubToken:         token,
+
+		NoForks: !withForks,
 	}
+	f.addResourceLimits()
 
+	return f
+}
+
+func (f *envFile) addResourceLimits() {
 	// limit CPU for containers
-	gitbaseLimitCPU := "0.0"
-	gitcollectorLimitCPU := "0.0"
 	dockerCPUs, err := dockerNumCPU()
 	if err != nil { // show warning
 		fmt.Println(err)
 	}
 	// apply gitbase resource limits only when docker runs without any global limits
 	// it's default behaviour on linux
 	if runtime.NumCPU() == dockerCPUs {
-		gitbaseLimitCPU = fmt.Sprintf("%.1f", float32(dockerCPUs)-0.1)
+		f.GitbaseLimitCPU = float32(dockerCPUs) - 0.1
 	}
+	// always apply gitcollector limit
 	if dockerCPUs > 0 {
-		halfCPUs := float32(dockerCPUs)/2.0
+		halfCPUs := float32(dockerCPUs) / 2.0
+		// let container consume more than a half if there is only one cpu available
+		// otherwise it will be too slow
 		if halfCPUs < 1 {
 			halfCPUs = 1
 		}
-		gitcollectorLimitCPU = fmt.Sprintf("%.1f", halfCPUs-0.1)
+		f.GitcollectorLimitCPU = halfCPUs - 0.1
 	}
 
 	// limit memory for containers
-	gitbaseLimitMem := "0"
 	dockerMem, err := dockerTotalMem()
 	if err != nil { // show warning
 		fmt.Println(err)
 	}
 	// apply memory limits only when only when docker runs without any global limits
 	// it's default behaviour on linux
 	if dockerMem == memory.TotalMemory() {
-		gitbaseLimitMem = strconv.FormatUint(uint64(float64(dockerMem)*0.9), 10)
+		f.GitbaseLimitMem = uint64(float64(dockerMem) * 0.9)
+	}
+}
+
+var newlineChar = "\n"
+
+func init() {
+	if runtime.GOOS == "windows" {
+		newlineChar = "\r\n"
+	}
+}
+
+// implementation can be moved to separate package if we need to marshal any other structs
+// supports only simple types
+func (f envFile) MarshalEnv() ([]byte, error) {
+	var b bytes.Buffer
+
+	v := reflect.ValueOf(f)
+	rType := v.Type()
+
+	for i := 0; i < rType.NumField(); i++ {
+		field := rType.Field(i)
+		fieldEl := v.Field(i)
+		if field.Anonymous {
+			panic("struct composition isn't supported")
+		}
+
+		name := nameToVar(field.Name)
+		switch field.Type.Kind() {
+		case reflect.Slice:
+			slice := make([]string, fieldEl.Len())
+			for i := 0; i < fieldEl.Len(); i++ {
+				slice[i] = fmt.Sprintf("%v", fieldEl.Index(i).Interface())
+			}
+			fmt.Fprintf(&b, "%s=%v%s", name, strings.Join(slice, ","), newlineChar)
+		case reflect.Bool:
+			// marshal false value as empty string instead of "false" string
```

Comment:
I think it's not a problem with the current env vars, but is it possible that in the future we'll need to write `VARIABLE=false`?

Reply:
Empty string for this, in general, makes more sense.
If we need to write `false` it should be `string` type, not bool then most probably.